aws-storage-s3/src/androidTest/java/com/amplifyframework/storage/s3/AWSS3StorageUploadTest.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.storage.s3;
+
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.hub.HubChannel;
+import com.amplifyframework.hub.HubEvent;
+import com.amplifyframework.hub.SubscriptionToken;
+import com.amplifyframework.storage.StorageAccessLevel;
+import com.amplifyframework.storage.operation.StorageUploadFileOperation;
+import com.amplifyframework.storage.options.StorageUploadFileOptions;
+import com.amplifyframework.testutils.Sleep;
+
+import com.amazonaws.mobileconnectors.s3.transferutility.TransferState;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/**
+ * Instrumentation test for operational work on upload.
+ */
+public final class AWSS3StorageUploadTest extends StorageInstrumentationTestBase {
+
+    // TODO: This is a temporary work-around to resolve a race-condition
+    // TransferUtility crashes if a transfer is paused and instantly resumed.
+    private static final int SLEEP_DURATION_IN_MILLISECONDS = 300;
+
+    private static final StorageAccessLevel DEFAULT_ACCESS_LEVEL = StorageAccessLevel.PUBLIC;
```

Comment:
I this coded somewhere in the source? Can you do `@VisibleForTesting` on it, and re-use it here in the test code? Even if this becomes a `public` field in source, that's probably not the end of the world. If a user finds out the default is `PUBLIC`: so what.

Reply:
Oh i meant default for testing. It could have been anything, but I didn't want to sign in for testing functionality since access level was already tested in a separate class.